spec/lang/step/intrinsics.md

```diff
@@ -67,7 +67,7 @@ impl<M: Memory> Machine<M> {
         }
 
         let ptr = self.intptrcast.int2ptr(addr)?;
-        ret(Value::Ptr(ptr))
+        ret(Value::Ptr(ptr.widen(None)))
```

Comment:
So here you are assuming that `ret_ty` is a pointer to a *sized* type. That would have to be checked above.